src/Common/src/Interop/Windows/mincore/Interop.GetFullPathNameW.cs

```diff
@@ -22,19 +22,25 @@ partial class mincore
         [DllImport(Libraries.CoreFile_L1, EntryPoint = "GetFullPathNameW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
         private static extern int GetFullPathNamePrivate(string path, int numBufferChars, [Out]StringBuilder buffer, IntPtr mustBeZero);
 
-        internal static int GetFullPathName(string path, int numBufferChars, [Out]StringBuilder buffer, IntPtr mustBeZero)
+        internal static int GetFullPathName(string path, int numBufferChars, [Out]StringBuilder buffer)
         {
-            bool wasExtended = PathInternal.IsExtended(path);
-            if (!wasExtended)
-            {
-                path = PathInternal.EnsureExtendedPrefixOverMaxPath(path);
-            }
-            int result = GetFullPathNamePrivate(path, buffer.Capacity, buffer, mustBeZero);
+            // GetFullPathName is backwards from most APIs. It handles long "DOS" paths without a problem. While it understands
+            // extended paths (\\?\), it roots them incorrectly, eating into the volume name with relative paths.
+
+            PathInternal.ExtendedType extendedType = PathInternal.RemoveExtendedPrefix(ref path);
+            int result = GetFullPathNamePrivate(path, buffer.Capacity, buffer, IntPtr.Zero);
 
-            if (!wasExtended)
+            if (extendedType.HasFlag(PathInternal.ExtendedType.Extended))
```

Comment:
> 

FWIW, this doesn't really have much to do with the compiler.  HasFlag as implemented at https://github.com/dotnet/coreclr/blob/4cf8a6b082d9bb1789facd996d8265d3908757b2/src/mscorlib/src/System/Enum.cs#L868-L877 and https://github.com/dotnet/coreclr/blob/4cf8a6b082d9bb1789facd996d8265d3908757b2/src/vm/reflectioninvocation.cpp#L3707-L3747 is simply much more expensive than a bit-wise-and plus a comparison.